### PR TITLE
Skip `__construct()` in BoolReturnTypeFromStrictScalarReturnsRector

### PR DESCRIPTION
**Pull request by staabm**

### CODE REVIEW

File: rules-tests/TypeDeclaration/Rector/ClassMethod/BoolReturnTypeFromStrictScalarReturnsRector/Fixture/skip_construct.php.inc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\BoolReturnTypeFromStrictScalarReturnsRector\Fixture;
+
+final class SkipConstructor
+{
+    public function __construct()
+    {
+        return true;
```
Comment (staabm) [inline rules-tests/TypeDeclaration/Rector/ClassMethod/BoolReturnTypeFromStrictScalarReturnsRector/Fixture/skip_construct.php.inc:9]: I am aware that having a `__construct` method and returning something in it does not actually make sense, but I stumbled over such a case today, and rector was adding a `:bool` return type on said method - which leads to a fatal error at runtime

**Comment by samsonasik** [comment]
You can use `ClassMethodReturnTypeOverrideGuard` like other rule

**Comment by staabm** [comment]
great catch, thanks

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Thank you @staabm

**Comment by TomasVotruba** [comment]
Thanks :clap:

**Pull request merged**